shared_model/builders/protobuf/common_objects/proto_account_asset_builder.hpp

```diff
@@ -48,12 +48,9 @@ namespace shared_model {
       }
 
       AccountAssetBuilder balance(const interface::Amount &amount) {
-        // TODO: 14.02.2018 nickaleks add proper amount initialization IR-972
         AccountAssetBuilder copy(*this);
-
-        auto *amount_proto = new iroha::protocol::Amount();
-        amount_proto->mutable_value()->set_first(
-            amount.intValue().template convert_to<uint64_t>());
+        auto amount_proto = new iroha::protocol::Amount();
```

Comment:
It's better defer allocation stuff to protobuf, use `mutable_balance` instead `set_allocated_balance`